deltachat-ffi/deltachat.h

```diff
@@ -3240,6 +3276,36 @@ int             dc_msg_is_setupmessage        (const dc_msg_t* msg);
 char*           dc_msg_get_setupcodebegin     (const dc_msg_t* msg);
 
 
+/**
+ * Check if the message is a videochat invitation.
+ *
+ * Typically, such messages are rendered differently by the UIs,
+ * if so, they should contain a button to join the videochat.
+ * The url for joining can be retrieved using dc_msg_get_videochat_url().
+ *
+ * @param msg The message object.
+ * @return Tristate, 0=message is no videochat invitation,
+ *     1=message is a videochat invitation that should be handled by a supported browser,
+ *     2=message is a videochat invitation that can be handled internally or by a supported browser.
+ */
+int dc_msg_is_videochat_invitation (const dc_msg_t* msg);
```

Comment:
On reception, we always do `dc_msg_get_viewtype` to determine how the message is displayed. Now we need to also call `dc_msg_is_videochat_invitation` before that? Shouldn't invitation just be another viewtype?

Reply:
well, tbh, i am also unsure about that bit :)

Reply:
> was looking a bit at dc_msg_is_securejoin() here :)

You mean `dc_msg_is_setupmessage`?

Reply:
i just had a look at android/ios, using a new message-type would probably zero-effort, so, indeed, maybe this is more prober.

also as we also have different types for eg. AUDIO/VOICE and VIDEO/GIF, putting TEXT and CALL to one type is maybe a bit stingy :)

Reply:
using a dedicated type now.